turbo/stages/stageloop.go

```diff
@@ -40,70 +40,68 @@ func StageLoop(
 	updateHead func(ctx context.Context, head uint64, hash common.Hash, td *uint256.Int),
 	waitForDone chan struct{},
 	loopMinTime time.Duration,
-) {
+	initialCycle bool,
+) (bool, bool) {
 	defer close(waitForDone)
-	initialCycle := true
 
-	for {
```

Comment:
`for` is removed, so it is not a "Loop" anymore?

Reply:
Oh, I see, you moved the loop out of this function

Reply:
Can you please modify the loop here instead of taking it out of the function? You have `hd` pointer passed here, so you can call `IsLinkQueueEmpty()` here too

Reply:
Should I add the loop inside of the function and outside of the function too? I am added the loop outside to keep checking wether or not we run `stageLoop` constantly.